cw_bitcoin/lib/electrum_transaction_history.dart

```diff
@@ -38,11 +39,26 @@ abstract class ElectrumTransactionHistoryBase
       transactions.forEach((_, tx) => _updateOrInsert(tx));
 
   @override
-  Future<void> save() async {
+  Future<void> save({Map<String, TransactionDirection>? txsDirection}) async {
     try {
       final dirPath =
           await pathForWalletDir(name: walletInfo.name, type: walletInfo.type);
       final path = '$dirPath/$_transactionsHistoryFileName';
+
+      if (txsDirection != null) {
+        transactions.forEach((key, value) {
+          txsDirection.forEach((directionKey, directionValue) {
+            if (directionKey == key) {
+              value.direction = directionValue;
+            }
+          });
+          if (!txsDirection.containsKey(key)) {
+            value.direction = TransactionDirection.outgoing;
+          }
+          ;
+        });
```

Comment:
no need for the nested `forEach` you can just set the direction of the transaction with the `txsDirection[key]` and if null set it with the outgoing

`

       transactions.forEach((key, value) {
            value.direction = txsDirection[key] ?? TransactionDirection.outgoing;
        });